test/blocklib/algorithm/fft/fft_common.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef GRAPH_PROTOTYPE_ALGORITHM_FFT_COMMON_HPP
+#define GRAPH_PROTOTYPE_ALGORITHM_FFT_COMMON_HPP
+
+#include "fft_types.hpp"
+#include <algorithm>
+#include <cmath>
+#include <vector>
+
+namespace gr::algorithm {
+
+template<ComplexType T, typename U>
+void
+computeMagnitudeSpectrum(const std::vector<T> &fftOut, std::vector<U> &magnitudeSpectrum, std::size_t fftSize, bool outputInDb) {
+    // TODO: check vector sizes
+    using PrecisionType = typename T::value_type;
+    std::transform(fftOut.begin(), std::next(fftOut.begin(), magnitudeSpectrum.size()), magnitudeSpectrum.begin(), [fftSize, outputInDb](const auto &c) {
+        const auto mag{ std::hypot(c.real(), c.imag()) * PrecisionType(2.0) / static_cast<PrecisionType>(fftSize) };
+        return static_cast<U>(outputInDb ? PrecisionType(20.) * std::log10(std::abs(mag)) : mag);
+    });
+}
+
+template<ComplexType T, typename U>
+void
+computePhaseSpectrum(const std::vector<T> &fftOut, std::vector<U> &phaseSpectrum, bool outputInDb) {
+    // TODO: check vector sizes
+    using PrecisionType = typename T::value_type;
+    std::transform(fftOut.begin(), std::next(fftOut.begin(), phaseSpectrum.size()), phaseSpectrum.begin(), [outputInDb](const auto &c) {
+        const auto phase{ std::atan2(c.imag(), c.real()) };
+        return static_cast<U>(outputInDb ? PrecisionType(20.) * std::log10(std::abs(phase)) : phase);
```

Comment:
Output in db doesn't make sense for phase... rather radians vs. degree. This should also include the (optional) unwrapping of the phase to have a linear definition.

... TODO should be handled. ;-)